gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ManagedHttpJsonChannel.java

```diff
@@ -84,10 +85,16 @@ public <RequestT, ResponseT> HttpJsonClientCall<RequestT, ResponseT> newCall(
 
   @Override
   public synchronized void shutdown() {
+    // Calling shutdown() twice should no-op
     if (isTransportShutdown) {
       return;
     }
     try {
+      // Only shutdown the executor if it was created by Gax. External executors
+      // should be managed by the user.
+      if (usingDefaultExecutor) {
+        ((ExecutorService) executor).shutdown();
```

Comment:
The defaultExecutor's type *currently* is always set to be an ExecutorService since it's created via `InstantiatingExecutorProvider`: https://github.com/googleapis/sdk-platform-java/blob/cc5724f04722e3d60eb9b4e02ecc1bdf5dd7a7fa/gax-java/gax/src/main/java/com/google/api/gax/core/InstantiatingExecutorProvider.java#LL62C3-L62C3

I can add a quick check with `instanceOf` to ensure that it won't error out. I think changing the type from Executor to ExecutorService impacts a bunch of core classes (TransportChannelProvider, InstantiatingGrpcChannelProvider, InstantiatingHttpJsonChannelProvider)